### DISCUSSION TITLE
Auto incrementing on transaction rollback

### DISCUSSION THREAD

**Discussion by szattila98** [SeaQL/sea-orm#860] [Q&A] [answered]
Hello, with the `begin & commit / rollback` transaction API, if the rollback happens after an insertion, the primary key auto increment is not rolled back. So next time when an insert succeeds, the primary key will be +2 than the last one. Is this the expected behavior? If so, why?

Example pseudocode:
last entity in db has id 9
**First insertion**
insert(entity) -> id would be 10
rollback -> entity not persisted
**Second insertion**
insert(entity) -> next id should be 10 because there isn't an entity in the db with id 10
commit -> entity persisted, but the id is 11

I didn't check if the closure transaction API has the same behaviour but I presume it does.

**Comment by billy1624** [comment, marked as answer]
Hey @szattila98, sorry for the delay. This behaviour doesn't defined by SeaORM instead it's how database handled insert during transaction. You can find the explanation online, e.g. https://stackoverflow.com/questions/449346/mysql-auto-increment-does-not-rollback

**Reply by szattila98** [reply]
Thank you very much, it makes complete sense.
